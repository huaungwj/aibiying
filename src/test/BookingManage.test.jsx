import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import TestBookingManage from "../components/BookingManage/BookingManage";
// import {} from "enzy"

afterEach(cleanup);

it("should render the bookings page", () => {
    const { container, getByTestId } = render(<TestBookingManage />);
    const bookingBox = getByTestId("bookingBox");
    const pageTitle = getByTestId("page-title");

    expect(container.innerHTML).toMatch(
        "Please manage your listing information!"
    );
    expect(bookingBox).toContainElement(pageTitle);
});

it("There should be mybookings text", () => {
    const { getByTestId } = render(<TestBookingManage />);
    expect(getByTestId("page-title")).toHaveTextContent("My Bookings");
});

it("should take a snapshot", () => {
    const { asFragment } = render(<TestBookingManage />);
    expect(asFragment()).toMatchSnapshot();
});
