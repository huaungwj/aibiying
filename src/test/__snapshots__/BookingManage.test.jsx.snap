// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="aby_container"
    data-testid="bookingBox"
    style="margin-top: 80px; font-size: 32px;"
  >
    <section
      id="userBookings"
    >
      <h1
        class="page-title"
        data-testid="page-title"
      >
        My Bookings
      </h1>
      <p>
        Please manage your listing information! 
      </p>
      <div
        class="row card-deck"
        style="display: flex; flex: 1; flex-wrap: wrap; justify-content: flex-start;"
      />
    </section>
  </div>
</DocumentFragment>
`;
